--- CMakeLists.txt
+++ CMakeLists.txt
@@ -600,12 +600,6 @@ macro(common_libktx_settings target enab
                 lib/internalexport_mingw.def
                 $<${enable_write}:lib/internalexport_write_mingw.def>
             )
-            # Need these flags if mingw happens to target the ucrt (new) rather
-            # than the legacy msvcrt. Otherwise tests will fail to run because
-            # the necessary dlls will be missing. If we statically link
-            # them instead it's fine. This does not cause any abberations if
-            # the mingw toolchain targets msvcrt instead.
-            target_link_options(${target} PUBLIC -static-libgcc -static-libstdc++)
         else()
             target_sources(
                 ${target}
--- lib/astc_encode.cpp
+++ lib/astc_encode.cpp
@@ -39,7 +39,7 @@
 #else
 // Provide pthreads support on windows
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
+#include <windows.h>
 
 typedef HANDLE pthread_t;
 typedef int pthread_attr_t;
