--- CMakeLists.txt
+++ CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.20)
 
 if(POLICY CMP0077)
     cmake_policy(SET CMP0077 NEW)
@@ -22,7 +22,13 @@ option(DOCTEST_WITH_MAIN_IN_STATIC_LIB  "Build a static lib (cmake target) with
 option(DOCTEST_NO_INSTALL  "Skip the installation process" OFF)
 option(DOCTEST_USE_STD_HEADERS  "Use std headers" OFF)
 
-add_library(${PROJECT_NAME} INTERFACE)
+add_library(${PROJECT_NAME} STATIC doctest/parts/doctest.cpp)
+target_compile_definitions(${PROJECT_NAME} PRIVATE DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN DOCTEST_CONFIG_COLORS_ANSI)
+
+find_package(fmt REQUIRED)
+target_link_libraries(${PROJECT_NAME} PUBLIC fmt::fmt)
+target_compile_definitions(${PROJECT_NAME} PUBLIC DOCTEST_CONFIG_USE_STD_HEADERS)
+
 add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
 
 if(NOT CMAKE_VERSION VERSION_LESS 3.8)
--- doctest/parts/doctest.cpp
+++ doctest/parts/doctest.cpp
@@ -2886,15 +2886,9 @@ namespace {
         }
 
         void printVersion() {
-            if(opt.no_version == false)
-                s << Color::Cyan << "[doctest] " << Color::None << "doctest version is \""
-                  << DOCTEST_VERSION_STR << "\"\n";
         }
 
         void printIntro() {
-            printVersion();
-            s << Color::Cyan << "[doctest] " << Color::None
-              << "run with \"--" DOCTEST_OPTIONS_PREFIX_DISPLAY "help\" for options\n";
         }
 
         void printHelp() {
@@ -3086,7 +3080,6 @@ namespace {
         void test_run_start() override { printIntro(); }
 
         void test_run_end(const TestRunStats& p) override {
-            separator_to_stream();
             s << std::dec;
 
             auto totwidth = int(std::ceil(log10((std::max(p.numTestCasesPassingFilters, static_cast<unsigned>(p.numAsserts))) + 1)));
--- doctest/parts/doctest_fwd.h
+++ doctest/parts/doctest_fwd.h
@@ -394,6 +394,7 @@ DOCTEST_GCC_SUPPRESS_WARNING_POP
 #include <iosfwd>
 #include <cstddef>
 #include <ostream>
+#include <fmt/format.h>
 #else // DOCTEST_CONFIG_USE_STD_HEADERS
 
 #if DOCTEST_CLANG
@@ -846,7 +847,11 @@ namespace detail {
     struct StringMakerBase
     {
         template <typename T>
-        static String convert(const DOCTEST_REF_WRAP(T)) {
+        static String convert(const T& v) {
+            if constexpr (std::is_constructible_v<fmt::formatter<std::remove_cvref_t<T>>>) {
+              const std::string str = fmt::format("{}", v);
+              return { str.data(), static_cast<unsigned>(str.size()) };
+            }
             return "{?}";
         }
     };
