diff --git lib/lz4.h lib/lz4.h
index 7ab1e48..b58af6c 100644
--- lib/lz4.h
+++ lib/lz4.h
@@ -73,28 +73,26 @@ extern "C" {
   The `lz4` CLI can only manage frames.
 */
 
-/*^***************************************************************
+/****************************************************************
 *  Export parameters
 *****************************************************************/
-/*
-*  LZ4_DLL_EXPORT :
-*  Enable exporting of functions when building a Windows DLL
-*  LZ4LIB_VISIBILITY :
-*  Control library symbols visibility.
-*/
-#ifndef LZ4LIB_VISIBILITY
-#  if defined(__GNUC__) && (__GNUC__ >= 4)
-#    define LZ4LIB_VISIBILITY __attribute__ ((visibility ("default")))
+
+#ifdef _WIN32
+#  define LZ4LIB_EXPORT __declspec(dllexport)
+#  define LZ4LIB_IMPORT __declspec(dllimport)
+#else
+#  define LZ4LIB_EXPORT __attribute__((__visibility__("default")))
+#  define LZ4LIB_IMPORT
+#endif
+
+#ifdef LZ4LIB_SHARED
+#  ifdef LZ4LIB_EXPORTS
+#    define LZ4LIB_API LZ4LIB_EXPORT
 #  else
-#    define LZ4LIB_VISIBILITY
+#    define LZ4LIB_API LZ4LIB_IMPORT
 #  endif
-#endif
-#if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
-#  define LZ4LIB_API __declspec(dllexport) LZ4LIB_VISIBILITY
-#elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
-#  define LZ4LIB_API __declspec(dllimport) LZ4LIB_VISIBILITY /* It isn't required but allows to generate better code, saving a function pointer load from the IAT and an indirect jump.*/
 #else
-#  define LZ4LIB_API LZ4LIB_VISIBILITY
+#  define LZ4LIB_API
 #endif
 
 /*------   Version   ------*/
