diff --git i/source/common/unicode/utypes.h w/source/common/unicode/utypes.h
index f890d5d..afedf55 100644
--- i/source/common/unicode/utypes.h
+++ w/source/common/unicode/utypes.h
@@ -136,9 +136,9 @@
  * ICU 1.8.x on EBCDIC, etc..
  * @stable ICU 2.0
  */
-#define U_ICUDATA_NAME    "icudt" U_ICU_VERSION_SHORT U_ICUDATA_TYPE_LETTER
+#define U_ICUDATA_NAME    "icudt" U_ICUDATA_TYPE_LETTER
 #ifndef U_HIDE_INTERNAL_API
-#define U_USRDATA_NAME    "usrdt" U_ICU_VERSION_SHORT U_ICUDATA_TYPE_LETTER  /**< @internal */
+#define U_USRDATA_NAME    "usrdt" U_ICUDATA_TYPE_LETTER  /**< @internal */
 #define U_USE_USRDATA     0  /**< @internal */
 #endif  /* U_HIDE_INTERNAL_API */
 
