diff --git i/source/glbinding-aux/CMakeLists.txt w/source/glbinding-aux/CMakeLists.txt
index 44d9bd9..7e88400 100644
--- i/source/glbinding-aux/CMakeLists.txt
+++ w/source/glbinding-aux/CMakeLists.txt
@@ -55,16 +55,11 @@ set(headers
     ${include_path}/ValidVersions.h
 
     ${include_path}/debug.h
-    ${include_path}/logging.h
 
-    ${include_path}/types_to_string.h
-    ${include_path}/types_to_string.inl
-    
     # KHR binding
     ${include_path}/RingBuffer.h
     ${include_path}/RingBuffer.inl
     ${include_path}/ValidVersions.h
-    ${include_path}/types_to_string.h
 )
 
 # add featured headers
@@ -94,15 +89,8 @@ set(sources
     ${source_path}/ValidVersions_list.cpp
 
     ${source_path}/debug.cpp
-    ${source_path}/logging.cpp
-    ${source_path}/logging_private.h
 
-    ${source_path}/types_to_string.cpp
-    ${source_path}/types_to_string_private.cpp
-    ${source_path}/types_to_string_private.h
-    
     # KHR binding
-    ${source_path}/types_to_string.cpp
     ${source_path}/ValidVersions.cpp
 )
 
