diff --git i/source/glbinding/source/getProcAddress.cpp w/source/glbinding/source/getProcAddress.cpp
index 1a5027e..595f7ed 100644
--- i/source/glbinding/source/getProcAddress.cpp
+++ w/source/glbinding/source/getProcAddress.cpp
@@ -34,7 +34,7 @@ ProcAddress getProcAddress(const char * name)
 		return reinterpret_cast<ProcAddress>(procAddress);
 	}
 
-	procAddress = ::GetProcAddress(module, name);
+	procAddress = reinterpret_cast<decltype(procAddress)>(::GetProcAddress(module, name));
     return reinterpret_cast<ProcAddress>(procAddress);
 }
 
