--- mingw-w64-headers/include/psdk_inc/intrin-impl.h
+++ mingw-w64-headers/include/psdk_inc/intrin-impl.h
@@ -1,3 +1,6 @@
+#ifndef __INTRINSIC_DEFINED___cpuidex
+#define __INTRINSIC_DEFINED___cpuidex
+#endif
 /**
  * This file has no copyright assigned and is placed in the Public Domain.
  * This file is part of the mingw-w64 runtime package.
--- mingw-w64-headers/crt/wchar.h
+++ mingw-w64-headers/crt/wchar.h
@@ -40,6 +40,9 @@ extern "C" {
 
 #ifndef _FILE_DEFINED
   struct _iobuf {
+#ifdef _UCRT
+    void *_Placeholder;
+#else
     char *_ptr;
     int _cnt;
     char *_base;
@@ -48,6 +51,7 @@ extern "C" {
     int _charbuf;
     int _bufsiz;
     char *_tmpfname;
+#endif
   };
   typedef struct _iobuf FILE;
 #define _FILE_DEFINED
